great_expectations/experimental/datasources/interfaces.py

```diff
@@ -1,7 +1,9 @@
 from __future__ import annotations
 
 import dataclasses
+import itertools
```

Comment:
The changes in this file are mostly pulling functionality up from `sql_datasources.py`. I will highlight any changes that happened beyond moving blocks of code.